src/OSPSuite.Core/Domain/Services/ObservedDataTask.cs

```diff
@@ -73,6 +73,9 @@ public bool Delete(IEnumerable<DataRepository> observedDataToBeRemoved, bool sil
       {
          var observedDataToRemoveList = observedDataToBeRemoved.ToList();
 
+         if (observedDataToRemoveList.Count < 1) 
```

Comment:
use Any() for this. Is the list empty or not basically

Reply:
and while you are here, can you check for null? 

e.g. var observedDataToRemoveList = observedDataToBeRemoved?.ToList();

if(observedDataToRemoveList ==null || !observedDataToRemoveList.Any())
    return true

This should never happen but who knows
And add a test for this